src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -697,6 +766,9 @@ void ImageReplayer<I>::on_stop_local_image_close_finish(int r)
   m_replay_status_formatter = nullptr;
 
   m_remote_journaler->stop_replay();
+  if (m_stopping_for_resync) {
+    m_remote_journaler->unregister_client();
```

Comment:
Note: under PR #9309 I converted the shut down path to be fully asynchronous to fix an issue.  If we need to unregister the client in the shut down path, we need this to be async as well.  However, I am wondering what would happen if rbd-mirror crashed between unregistering with the remote journal and performing the delete.